### ISSUE TITLE
FusorSV for GRCh38 Data

### ISSUE THREAD

**Issue by StevenIg** [timothyjamesbecker/FusorSV#11]
Hello,

I want to use FusorSV on Data that align and calls on GRCh38. But Your models is in GRCh37. It's possible to use FusorSV in that case? 
The -L option is a liftover for your models or for the output vcf files? 

Thank you in advance,

Steven

**Comment by timothyjamesbecker** [comment]
The 1000 genomes phase 3 reference (it is a modified GRCh37 variant that includes a large decoy sequence) was used to train the current human_g1k_v37_decoy model and it should work fine with GRCh38 since it is close in general performance (GRCh38 has a higher mapping rate in general which means slightly better SV calling).  A few of the phase 3 calls are omitted in the GRCh38 coordinate space since the 27 populations seemed to all have specific calls against GRCh37.  You should not use the liftover option if you are starting with GRCh38 vcf calls, as this will change all of your coordinates!

**Comment by StevenIg** [comment]
Thanks you for your reply.

 I run fusorSV (from https://github.com/TheJacksonLaboratory/SVE) but vcf results files are empty.

I use this command line : 
condaPython ${PATH_SVE}/scripts/FusorSV/FusorSV.py -f ${PATH_SVE}//scripts/FusorSV/data/models/default.pickle -r ${FILE_REF_FASTA} -i ${PATH_WORKDIR_DIR}/vcfFiles/ -p 8 -o ${PATH_WORKDIR_DIR}/resultsSV -c chr1,chr2,chr3,chr4,chr5,chr6,chr7,chr8,chr9,chr10,chr11,chr12,chr13,chr13,chr14,chr15,chr16,chr17,chr18,chr19,chr20,chr21,chr22,chrX,chrY,chrM

I use Delly and Lumpy for 3 samples with my script and not with sve call.
Genome is GCRh38.

thank you,

Steven

**Comment by timothyjamesbecker** [comment]
If you are not getting calls from Lumpy and Delly it could either be a problem with your BAM files or an issue with the SVE framework which is running several callers.  You should file an issue at https://github.com/TheJacksonLaboratory/SVE if you are using that version, as I do not maintain those sources (even though I authored them).  Unfortunately finding flaws with BAM files can be time consuming and always requires some troubleshooting and exploration of the tags inside each header.  Some common issues arise with improperly setup read groups (some discussion here: https://gatkforums.broadinstitute.org/gatk/discussion/6472/read-groups).  A great way to do this is to download the header from one of the files used and tested with the SVE/FusorSV framework from the 1000 genomes site.  These files are known to be correct and have the BAM indexes and so a newer version of samtools can download the header only which is very fast:

samtools view -SH ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/data/HG00096/high_coverage_alignment/HG00096.wgs.ILLUMINA.bwa.GBR.high_cov_pcr_free.20140203.bam

**Comment by MaestSi** [comment]
Hello,
I tried running FusorSV on Homo_sapiens_assembly38.fasta too, and tried using docker.
I would like to run it on NA12878_S10.vcf, NA12878_S11.vcf, NA12878_S18.vcf, NA12878_S35.vcf and NA12878_S4.vcf VCF files, obtained with SVE and stored in /mnt/cifs01/simone/NA12878/test_FusorSV/VCF_files/NA12878
This is the command line I used:
```
CHROMS='chr1,chr2,chr3,chr4,chr5,chr6,chr7,chr8,chr9,chr10,chr11,chr12,chr13,chr14,chr15,chr16,chr17,chr18,chr19,chr20,chr21,chr22,chrX,chrY,chrM'
EXCLUDE='1,9,13,14,17,36,38'
MNT_INPUT_DIR=/data
docker run -v /mnt/cifs01/simone/NA12878:$MNT_INPUT_DIR -it timothyjamesbecker/fusorsv FusorSV.py -r $MNT_INPUT_DIR/Homo_sapiens_assembly38.fasta \
-i $MNT_INPUT_DIR/test_FusorSV/VCF_files/ -f DEFAULT -M 0.5 \
-o $MNT_INPUT_DIR/test_FusorSV/FusorSV_docker_output -p 10 -c $CHROMS -E $EXCLUDE \
-m $MNT_INPUT_DIR/Homo_sapiens_assembly38.svmask.fasta_svmask.json
```

This is the output I got.

> no contig directory specified
> using stage id exclude list:[1, 9, 13, 14, 17, 36, 38]
> processing samples ['/data/test_FusorSV/VCF_files/NA12878']
>  for chroms ['chr1', 'chr2', 'chr3', 'chr4', 'chr5', 'chr6', 'chr7', 'chr8', 'chr9', 'chr10', 'chr11', 'chr12', 'chr13', 'chr14', 'chr15', 'chr16', 'chr17', 'chr18', 'chr19', 'chr20', 'chr21', 'chr22', 'chrX', 'chrY', 'chrM']
> making a new coordinate offset json file
> merging the svmask regions
> svmask regions merged in 0.0 sec
> reading, parsing, partitioning and writing sample VCFs
> reading sample NA12878
> finished reading 0 out of 1 samples generating 0 partitions in 0.42 sec
> apply fusion model to sample inputs and generating fusorSV ouput
> starting fusorSV discovery on sample NA12878
> finished reading samples in 114.37 sec
> cluster merging tool processing samples
> completed = True
> run 0 in 297.46 sec
> ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

However, the output VCF file is empty a part from the header.
I already asked for help on SVE GitHub page, but wanted to try docker version too.
Thank you very much

**Comment by timothyjamesbecker** [comment]
The most common issue here would be that of not keeping the same reference fasta used in alignment for the VCF files.  You cannot align FASTQ and create a BAM with one reference sequence and then use another one for your VCF files, since that is inconsistent.  As long as the same reference file is used to generate the BAMs and VCFs and then passed to FusorSV there should not be 0 partitions generated as show here:   "finished reading 0 out of 1 samples generating 0 partitions in 0.42 sec"
This indicates there is a mismatch in your data workflow.
